### ISSUE TITLE
Apple CA store does not trust Google's cert?

### ISSUE THREAD

**Issue by quantonganh** [nabla-c0d3/sslyze#75]
sslyze.py --version

```
0.11.0
```

sslyze.py --regular mail.google.com

```
  * Certificate - Trust:
      Hostname Validation:               OK - Subject Alternative Name matches
      Mozilla NSS CA Store (04/2015):    OK - Certificate is trusted
      Microsoft CA Store (04/2015):      OK - Certificate is trusted
      Java 6 CA Store (Update 65):       OK - Certificate is trusted
      Apple CA Store (OS X 10.10.3):     FAILED - Certificate is NOT Trusted: unable to get local issuer certificate
      Certificate Chain Received:        ['mail.google.com', 'Google Internet Authority G2', 'GeoTrust Global CA']
```

It still worked fine with version 0.9:

```
  * Certificate - Trust:
      Hostname Validation:               OK - Subject Alternative Name matches
      "Microsoft - 04/2014" CA Store:    OK - Certificate is trusted
      "Mozilla NSS - 01/2014" CA Store:  OK - Certificate is trusted
      "Apple - OS X 10.9.2" CA Store:    OK - Certificate is trusted
      "Java 6 - Update 65" CA Store:     OK - Certificate is trusted
```

There are some different between them:

```
apple.pem 0.9: 5617 lines, 341129 characters
apple.pem 0.11: 5382 lines, 327254 characters
```

**Comment by tam7t** [comment]
I'm having the same results with `sslyze` and getting the same error when using homebrew's `openssl` as well:
```bash
~ $ openssl s_client -connect fonts.googleapis.com:443 -verify 5
verify depth is 5
CONNECTED(00000003)
depth=2 /C=US/O=GeoTrust Inc./CN=GeoTrust Global CA
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=2 /C=US/O=GeoTrust Inc./CN=GeoTrust Global CA
verify error:num=27:certificate not trusted
verify return:1
depth=1 /C=US/O=Google Inc/CN=Google Internet Authority G2
verify return:1
depth=0 /C=US/ST=California/L=Mountain View/O=Google Inc/CN=*.googleapis.com
verify return:1
---
```

It does succeed when using the equifax ca pem:

```bash
~ $ openssl s_client -connect fonts.googleapis.com:443 -verify 5 -CAfile /usr/local/etc/openssl/certs/Equifax_Secure_Certificate_Authority.pem
verify depth is 5
CONNECTED(00000003)
depth=3 /C=US/O=Equifax/OU=Equifax Secure Certificate Authority
verify return:1
depth=2 /C=US/O=GeoTrust Inc./CN=GeoTrust Global CA
verify return:1
depth=1 /C=US/O=Google Inc/CN=Google Internet Authority G2
verify return:1
depth=0 /C=US/ST=California/L=Mountain View/O=Google Inc/CN=*.googleapis.com
verify return:1
---
```

Homebrew openssl says
```bash
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /usr/local/etc/openssl/certs

and run
  /usr/local/opt/openssl/bin/c_rehash
```
The equifax pem was already in my `openssl/certs` folder but running `c_rehash` does not change the output of `openssl`.

**Comment by nabla-c0d3** [comment]
Very interesting - indeed my guess was that the Equifax cert had been removed. 
If you try to validate the chain with openssl s_client and the GeoTrust certificate as the CA store, the validation still fails because OpenSSL does not stop at the GeoTrust cert and looks for the Equifax one (as explained in the link); this is exactly what I was seeing when trying to troubleshoot this. I didn't know this behavior was specific to OpenSSL but that explains why browsers give a different result when validating this chain.

**Comment by nabla-c0d3** [comment]
It seems like OpenSSL will fix this in 1.0.2b; when it gets released I can re-build SSLyze with this new version.

**Comment by nabla-c0d3** [comment]
Confirmed this was fixed in OpenSSL 1.0.2d (https://github.com/nabla-c0d3/nassl/commit/836ff18b300d8202a13339230345ef1e0bd210fd).
